pkg/buildkit/worker/pool.go

```diff
@@ -159,7 +160,16 @@ func (p *workerPool) Get(ctx context.Context) (string, error) {
 
 		// when the channel is closed, we receive nil
 		if result.pod != nil {
-			return p.buildEndpointURL(ctx, result.pod.Name)
+			addr, err := p.buildEndpointURL(ctx, result.pod.Name)
```

Comment:
maybe in a vacuum, `result` would have an `addr` instead of a pod and then the endpoint creation would happen during lease?

Reply:
I know there's reason why I passed around a `pod` instead of an `addr` in the past but that may no longer be necessary. I think this might be a relic from the beginnings of #30 but I'll need to check just to make sure. Looking at the code, passing the address seems viable which means we could do the lease and url building inside the reconciliation loop.

Reply:
I've added this to https://github.com/dominodatalab/hephaestus/pull/37 since I had to modify pod request/result passing there anyway.